homeassistant/components/tuya/climate.py

```diff
@@ -201,6 +209,8 @@ def target_temperature(self):
     @property
     def target_temperature_step(self):
         """Return the supported step of target temperature."""
+        if self._override_precision != PRECISION_DEFAULT:
+            return self._override_precision
```

Comment:
These days I have been thinking about this PR, `target_temperature` should be managed separately because could  have different value not compatible with precision.
You should remove this part and continue with this PR that in this case will not require TuyaHA library change. We will manage the `target_temperature `with a separate PR.

Reply:
But just thinking more ... why this is required? What happen if we just set precision to `PRECISION_TENTHS`? There are some undesired side effect? I think that we are making this problem more complex. Thoughts?

Reply:
My thermostat needs also to be tweaked on target temperature step. In fact, I can only set one unit. But my thermostat support 0.5 step.
With this override I can put also 0.5 step (e.g. 18.5 C). But I realized that I need also [this change in the library](https://github.com/PaulAnnekov/tuyaha/pull/59) because there's a `round(number)` that prevent me to set 0.5 values.

So with both these changes I can set 17.5C 😊.

There's a drawback, as you pointed out: what happens if you set PRECISION_TENTH and your thermostat support only HALVES? My thermostat puts the lower possible bound (target temperature 17.9 C -> it set 17.5C).

You should put the right precision on your thermostat. If you put higher precision, you're doing it wrong, but it's not a tragedy if all thermostats behave like mine (I obviously can't test it).

Reply:
I think that `precision` controls the visualization, `temperature_step` the step used to set  the target. This 2 value is not related, you can set `PRECISION_TENTH` for `precision` and `PRECISION_HALVES` (0.5) for `temperature_step` and everything will work as expected. So i think that leaving `PRECISION_TENTH` for `precision` is not an issue at all, it is only important to set proper value for `temperature_step`.
The changes to the library proposed by you can potentially break other devices that do not accept decimal value in the payload, so should be reviewed.
I already implemented in [Tuya Custom](https://github.com/ollo69/ha_tuya_custom) all required changes, in options page you can change independently `precision` and `temperature_step` and also contains local library changes to properly manage the set temperature command with decimal values. Please temporally replace standard integration with this custom component to perform test with all possible combination, than we will choose the correct change to implement.

Reply:
Thank you for the custom library. It is easier to test.
But I don't need it, if I understood correctly. 
We decided to add the precision override only in "read mode", so I revert the temperature step part and set temperature.
In that way, I don't need to change the external library, it's safer and easier to digest as pull request. 
Is that ok? 
If we agreed, I'll revert the changes and we fix in another pr the "set the temperature" part.

Reply:
Not exactly. My idea is that create an option to override `precision` is not required, because we can simply set to `PRECISION_TENTH` and forget it.
Than we need to fix `temperature_step`, and for this we need PRs for both TuyaHA and HA.
I already made changes in custom component, but I don't have a climate so I need confirmation that this solve the issue and work properly before pushing correct PR.

Reply:
I tried all possible things.

All worked well except these:

- If I don't check "Use divided Temperature value for set temperature command", the thermostat becomes unavailable and the logs say

```
Logger: custom_components.tuya_custom.tuyaha.tuyaapi
Source: custom_components/tuya_custom/tuyaha/tuyaapi.py:273
Integration: Tuya Custom (documentation)
First occurred: 5:50:03 PM (1 occurrences)
Last logged: 5:50:03 PM

control device error, error code is ValueOutOfRange
```
I have divider 2.

- If I select a different sensor in "Sensor for current temperature", it doesn't take the change. If I go to my lovelace thermostat I always see the thermostat temperature and not the sensor I put in the select box.

This behavior is OK for me, but it worth mentioning:

- If I put temperature_step to tenth even if my thermostat support halves, it set the first lower bound acceptable. (E.g 18.9 -> it sets 18.5). This behavior is the same as I got in my PR.

Reply:
Thanks for your test.
- About point 1, it is correct because is to manage different devices behavior. But probably I have to set default to True because seems the most frequent cases
- Point 2 is only for devices that do not provide current temperature info, and anyway I cannot apply this change in official component because it is not allowed
- Point 3 is ok, as you said is acceptable. Just wondering if tenth step make sense, but can be left if required, because we are not able to determinate the correct step from device payload.

I'm going to do some additional changes:
- I will set default to true for point 1 as stated before
- I will remove the option to change the precision, I think set it always to tenth is OK (less option, less confusion).
- I will try to manage possible temperature step as library option because I'm not sure that is allowed define step directly in HA.

It would be great if you will make again some test when new version will be released.

Reply:
Tested with the new version

- I confirm it's true by default "Use divided Temperature value for set temperature command".
- The integration is working as expected.

I had an issue in changing the divider: I changed the value, but the value on UI didn't reflect the changes. I restarted HASS and it worked again. In my log I had an issue with "limit login reached". Maybe it simply lost the connection and so he could not update the UI.

Now it's working

Reply:
Limit login can occur during 1st configuration. It is again a Tuya API limitation.
That's fine, I will start pushing PR with new changes to TuyaHA later tomorrow.